app/tests/verification_tests/test_admin.py

```diff
@@ -52,3 +78,71 @@ def test_deactivate_users(django_capture_on_commit_callbacks, settings):
     assert users[3].is_active is True
     assert users[3].verification.is_verified is True
     assert users[4].is_active is True
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    "perm_request_factory, perm_request_entity_attr, entity_factory",
+    [
+        (AlgorithmPermissionRequestFactory, "algorithm", AlgorithmFactory),
+        (ArchivePermissionRequestFactory, "archive", ArchiveFactory),
+        (
+            ReaderStudyPermissionRequestFactory,
+            "reader_study",
+            ReaderStudyFactory,
+        ),
+        (RegistrationRequestFactory, "challenge", ChallengeFactory),
+    ],
+)
+@pytest.mark.parametrize(
+    "access_request_handling, expected_request_status_without_verification, expected_request_status_with_verification",
+    [
+        (
+            AccessRequestHandlingOptions.ACCEPT_ALL,
+            RequestBase.ACCEPTED,
+            RequestBase.ACCEPTED,
+        ),
+        (
+            AccessRequestHandlingOptions.ACCEPT_VERIFIED_USERS,
+            RequestBase.PENDING,
+            RequestBase.ACCEPTED,
+        ),
+        (
+            AccessRequestHandlingOptions.MANUAL_REVIEW,
+            RequestBase.PENDING,
+            RequestBase.PENDING,
+        ),
+    ],
+)
+def test_verify_users_and_accept_pending_requests(
+    django_capture_on_commit_callbacks,
+    settings,
+    perm_request_factory,
+    perm_request_entity_attr,
+    entity_factory,
+    access_request_handling,
+    expected_request_status_without_verification,
+    expected_request_status_with_verification,
+):
+    settings.task_eager_propagates = (True,)
+    settings.task_always_eager = (True,)
+
+    usr = UserFactory()
+
+    t = entity_factory(access_request_handling=access_request_handling)
+    pr = perm_request_factory(**{"user": usr, perm_request_entity_attr: t})
+
+    VerificationFactory(user=usr, email_is_verified=True, is_verified=False)
+
+    assert pr.status == expected_request_status_without_verification
+
+    with django_capture_on_commit_callbacks(execute=True):
```

Comment:
Why is `django_capture_on_commit_callbacks` necessary here?

Reply:
Looking at it again, it is not necessary. There is no `on_commit` callbacks in the underlying function to justify its use.
Fixed it.

Reply:
Did you copy this test from somewhere? Is it also redundant there?

Reply:
I made tests earlier on the model level to test the save() of the model after adding the implementation ([test_models.py](https://github.com/comic/grand-challenge.org/blob/bc13992a12eef6985c2d3ca77025f3891703f755/app/tests/verification_tests/test_models.py#L75)). I used the same parametrized tests here with the only difference of calling mark_verified() from admin.py (the changed function) to test it here.
Should there be only a test in either in test_models.py or test_admin.py?